recipes/agent.rb

```diff
@@ -1,52 +1,41 @@
 include_recipe "zabbix::agent_#{node['zabbix']['agent']['install_method']}"
 include_recipe 'zabbix::agent_common'
 
-# Install configuration
-template 'zabbix_agentd.conf' do
-  path node['zabbix']['agent']['config_file']
-  source 'zabbix_agentd.conf.erb'
-  unless node['platform_family'] == 'windows'
+# Install configuration and optional additional agent config file containing UserParameter(s)
+case node['platform_family']
+when 'windows'
+  template 'zabbix_agentd.conf' do
+    path node['zabbix']['agent']['config_file']
+    source 'zabbix_agentd.conf.erb'
+    notifies :enable, 'service[Zabbix Agent]'
+    notifies :restart, 'service[Zabbix Agent]'
+    notifies :run, 'execute[config_firewall]'
+  end
+  template 'user_params.conf' do
+    path node['zabbix']['agent']['userparams_config_file']
+    source 'user_params.conf.erb'
+    notifies :enable, 'service[Zabbix Agent]'
+    notifies :restart, 'service[Zabbix Agent]'
+    notifies :run, 'execute[config_firewall]'
+    only_if { node['zabbix']['agent']['user_parameter'].length > 0 }
```

Comment:
Why does this template have this `only_if` block, but the above template does not?

Reply:
because user_parameter is optional, but config file is default